flang/lib/Lower/IntrinsicCall.cpp

```diff
@@ -2167,6 +2213,54 @@ IntrinsicLibrary::genRepeat(mlir::Type resultType,
       });
 }
 
+// RESHAPE
+fir::ExtendedValue
+IntrinsicLibrary::genReshape(mlir::Type resultType,
+                             llvm::ArrayRef<fir::ExtendedValue> args) {
+  assert(args.size() == 4);
+
+  // Handle source argument
+  auto source = builder.createBox(loc, args[0]);
+
+  // Handle shape argument
+  auto shape = builder.createBox(loc, args[1]);
+  fir::BoxValue shapeTmp = shape;
+  auto shapeRank = shapeTmp.rank();
+  assert(shapeRank == 1);
+  auto shapeTy = shape.getType();
+  auto shapeArrTy = fir::dyn_cast_ptrOrBoxEleTy(shapeTy);
+  auto resultRank = shapeArrTy.cast<fir::SequenceType>().getShape();
+
+  assert(resultRank[0] != fir::SequenceType::getUnknownExtent() &&
+         "shape arg must have constant size");
+
+  // Handle optional pad argument
+  auto pad = isAbsent(args[2])
+                 ? builder.create<fir::AbsentOp>(
+                       loc, fir::BoxType::get(builder.getI1Type()))
+                 : builder.createBox(loc, args[2]);
+
+  // Handle optional order argument
+  auto order = isAbsent(args[3])
+                   ? builder.create<fir::AbsentOp>(
+                         loc, fir::BoxType::get(builder.getI1Type()))
+                   : builder.createBox(loc, args[3]);
+
+  // Create mutable fir.box to be passed to the runtime for the result.
+  auto type = builder.getVarLenSeqTy(resultType, resultRank[0]);
+  auto resultMutableBox =
+      Fortran::lower::createTempMutableBox(builder, loc, type);
+
+  auto resultIrBox =
+      Fortran::lower::getMutableIRBox(builder, loc, resultMutableBox);
+
+  Fortran::lower::genReshape(builder, loc, resultIrBox, source, shape, pad,
+                             order);
+
+  return readAndAddCleanUp(resultMutableBox, resultType,
```

Comment:
Should it be type or resultType here?

Reply:
resultType. It's needed for the const Fortran::lower::SymbolBox::Intrinsic case.